graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/kotlin/ClientApiGenerator.kt

```diff
@@ -232,14 +232,15 @@ class KotlinClientApiGenerator(private val config: CodeGenConfig, private val do
 
     private fun createConcreteTypes(type: TypeDefinition<*>, javaType: TypeSpec.Builder, rootType: TypeSpec, prefix: String, processedEdges: Set<Pair<String, String>>, queryDepth: Int): KotlinCodeGenResult {
         return if (type is InterfaceTypeDefinition) {
+            val rootRef = if (javaType.build().name == rootType.name) "this" else "root"
```

Comment:
Can the use case be demonstrated with a test case? Maybe also add a comment about the need for this code for future reference

Reply:
We have a test case for this already. The existing test did not catch it because we were not compiling kotlin code.

Reply:
```
@Test
    fun interfaceReturnTypes() {
        val schema = """
            type Query {
                search(title: String): [Show]
            }
            
            interface Show {
                title: String
            }
            
            type Movie implements Show {
                title: String
                duration: Int
            }
            
            type Series implements Show {
                title: String
                episodes: Int
            }
        """.trimIndent()
        val codeGenResult = CodeGen(
            CodeGenConfig(
                schemas = setOf(schema),
                packageName = basePackageName,
                language = Language.KOTLIN,
                generateClientApi = true,
            )
        ).generate() as KotlinCodeGenResult
        assertThat(codeGenResult.queryTypes.size).isEqualTo(1)
        val typeSpec = codeGenResult.queryTypes[0].members[0] as TypeSpec
        assertThat(typeSpec.name).isEqualTo("SearchGraphQLQuery")
        assertThat(codeGenResult.clientProjections[0].name).isEqualTo("SearchProjectionRoot")
        val projectionType = codeGenResult.clientProjections[0].members[0] as TypeSpec
        assertThat(projectionType.funSpecs[0].name).isEqualTo("title")
        val movieProjectionType = codeGenResult.clientProjections[1].members[0] as TypeSpec
        assertThat(movieProjectionType.name).isEqualTo("Search_MovieProjection")
        val seriesProjectionType = codeGenResult.clientProjections[2].members[0] as TypeSpec
        assertThat(seriesProjectionType.name).isEqualTo("Search_SeriesProjection")

        compile(codeGenResult.clientProjections)
    }```